samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -0,0 +1,435 @@
+            headers.add("From: \"" + rt.getFrom() );
+        } else {
+            headers.add("From: \"" + fromName + "\" <" + fromEmail + ">" );
+        }
+        // Add a To: header of either the recipient (if only 1), or the sender (if multiple)
+        String              toName                  = fromName;
+        String              toEmail                 = fromEmail;
+
+        if (toAddress.size() == 1) {
+            User u = toAddress.get(0);
+            toName = u.getDisplayName();
+            toEmail = u.getEmail();
+        }
+
+        headers.add("To: \"" + toName + "\" <" + toEmail + ">" );
+
+        //SAK-21742 we need the rendered subject
+        headers.add("Subject: " + rt.getRenderedSubject());
+        headers.add("Content-Type: multipart/alternative; boundary=\"" + MULTIPART_BOUNDARY + "\"");
+        headers.add("Mime-Version: 1.0");
+        headers.add("Return-Path: <>");
+        headers.add("Auto-Submitted: auto-generated");
+
+        return headers;
+    }
+
+    // loadTemplate from ValidationLogicImpl.java
+    /**
+     * Load and register one or more email templates (contained in the given
+     * .xml file) with the email template service
+     *
+     * @param templateFileName - the name of the .xml file to load
+     * @param templateRegistrationString - the key (name) of the template to be saved to the service
+     */
+    @SuppressWarnings("unchecked")
+    private     void                loadTemplate                    (String templateFileName, String templateRegistrationString) {
+        log.info(this + " loading template " + templateFileName);
+
+        SecurityAdvisor yesMan = new SecurityAdvisor() {
+            public SecurityAdvice isAllowed( String userId, String function, String reference ) {
+                return SecurityAdvice.ALLOWED;
+            }
+        };
+
+        try {
+            // Push the yesMan SA on the stack and perform the necessary actions
+            securityService.pushAdvisor(yesMan);
+
+            // Load up the resource as an input stream
+            InputStream input = SamigoETSProviderImpl.class.getClassLoader().getResourceAsStream("" + templateFileName);
+            if ( input == null ) {
+                log.error( "Could not load resource from '" + templateFileName + "'. Skipping ..." );
+            } else {
+                // Parse the XML, get all the child templates
+                Document document = new SAXBuilder().build( input );
+                List<?> childTemplates = document.getRootElement().getChildren( "emailTemplate" );
+                Iterator<?> iter = childTemplates.iterator();
+
+                // Create and register a template with the service for each one found in the XML file
+                while ( iter.hasNext() ) {
+                    xmlToTemplate( (Element) iter.next(), templateRegistrationString );
+                }
+            }
+        } catch ( JDOMException e ) {
```

Comment:
If your going to apply the same logic to multiple exception types then combine them.